### PR TITLE
feat: `autorepeat-by-default` patch

### PR DESCRIPTION
**Pull request by TheJeterLP**

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import app.revanced.patcher.patch.annotations.Dependencies
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.layout.autorepeat.annotations.AutoRepeatCompatibility
+import app.revanced.patches.youtube.layout.autorepeat.fingerprints.AutoRepeatFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+
+@Patch
+@Dependencies(dependencies = [IntegrationsPatch::class])
+@Name("disable-auto-repeat")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt:18]: Rename to hide-auto-repeat-button for consistency across our patch bundle

File: src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Description("Disable auto repeat")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt:19]: Don't forget punctuation

File: src/main/kotlin/app/revanced/patches/youtube/layout/createbutton/patch/CreateButtonRemoverPatch.kt [hunk 3]
```diff
@@ -25,7 +25,7 @@ import org.jf.dexlib2.iface.reference.MethodReference
 @Dependencies(dependencies = [IntegrationsPatch::class, ResourceIdMappingProviderResourcePatch::class])
 @Name("disable-create-button")
 @Description("Disable the create button.")
-@CreateButtonCompatibility
+@AutoRepeatCompatibility
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtube/layout/createbutton/patch/CreateButtonRemoverPatch.kt:28]: This should not be committed

Comment (TheJeterLP) [reply src/main/kotlin/app/revanced/patches/youtube/layout/createbutton/patch/CreateButtonRemoverPatch.kt:28]: yes true

File: src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt (hunk 1 above)
Comment (TheJeterLP) [reply src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt:18]: but it does not hide a button. It disables auto repeat, theres no button for that

Comment (TheJeterLP) [reply src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt:18]: Yeah I saw that now too. It's there in Vanced too. Trying to figure out what Vanced actually did with that lol

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt]: Maybe "enable-autorepeat" is enough for the patch name?

Comment (TheJeterLP) [reply src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt]: I think people will get confused by that

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/patch/AutoRepeatPatch.kt]: Therefore a description is included. The name should be short and not responsible to completely descriptive. If you insist, maybe "autorepeat-by-default"

File: src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/fingerprints/AutoRepeatFingerprint.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+/*
+public final void ae() {
+        aq(aabj.ENDED);
+    }
+ */
+//Change method to:
+/*
+public final void ae() {
+    aq(aabj.ENDED);
+    app.revanced.integrations.sponsorblock.player.VideoInformation.videoEnded();
```
Comment (TheJeterLP) [inline src/main/kotlin/app/revanced/patches/youtube/layout/autorepeat/fingerprints/AutoRepeatFingerprint.kt:29]: @oSumAtrIX the SB Patch needs this method to be called. How do we do that? Should we leave it for this patch rn and then figure a way out later?

**Comment by TheJeterLP** [comment]
Closes as opening another PR from different base branch. See #91

**Pull request closed without merging**